### PR TITLE
Let's Swift 5.0 🚀

### PR DESCRIPTION
**Pull request by devxoul**
## TL;DR

* Migrate to Swift 5 grammar
* Update project settings
* Bump RxSwift dependency to 4.5.0

### CODE REVIEW

**Comment by kzaher** [comment]
Hi @devxoul ,

I think we need to cut a new release for 5.0. I'm fine with that. But we can't keep the same major version. Can you please update the major version?

**Comment by devxoul** [comment]
@kzaher, sure. Do you want me to include the version bump commit in this PR?

File: Sources/Differentiator/Diff.swift [hunk 1]
```diff
@@ -128,12 +128,14 @@ public enum Diff {
 
     private struct OptimizedIdentity<E: Hashable> : Hashable {
 
-        let hashValue: Int
         let identity: UnsafePointer<E>
 
         init(_ identity: UnsafePointer<E>) {
             self.identity = identity
-            self.hashValue = identity.pointee.hashValue
```
Comment (kzaher) [inline Sources/Differentiator/Diff.swift]: This was an optimization to memoize hashValue. let hashValue: Int should be saved regardless and used in func hash(into hasher: inout Hasher) {. This is just a copy of RxDataSources, but still ...

**Comment by kzaher** [comment]
@kzaher, sure. Do you want me to include the version bump commit in this PR? Yep, that would be great.

**Comment by devxoul** [comment]
@kzaher 

1.  Made Diff memoize `hashValue` with the property name `cachedHashValue` (since `hashValue` is reserved in `Hashable`)
2. Bumped version to 4.0.0

**Comment by pddkhanh** [comment]
@devxoul @kzaher Can you make this work with RxSwift 5 instead?

**Comment by kzaher** [comment]
@pddkhanh We will, need to release it first.

**Pull request merged**